File: linked lists/203. Remove Linked List Elements/removeElements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        curr = head

        while head and head.val == val:
            head = head.next if head.next else None
            curr = head
        
        while curr and curr.next:
            if curr.next.val == val:
                curr.next = curr.next.next if curr.next.next else None
            else:
                curr = curr.next
        
        return head

File: linked lists/203. Remove Linked List Elements/removeElements.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function removeElements(head: ListNode | null, val: number): ListNode | null {
    let curr = head;

    while (head && head.val === val) {
        head = head.next ? head.next : null;
        curr = head;
    }  

    while (curr && curr.next) {
        if (curr.next.val == val) {
            curr.next = curr.next.next ? curr.next.next : null;
        } else {
            curr = curr.next;
        }
    }

    return head;
};